resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -369,6 +371,7 @@ public DnsNameResolver(
             final DnsCache resolveCache,
             final DnsCnameCache cnameCache,
             final AuthoritativeDnsServerCache authoritativeDnsServerCache,
+            final String bindAddress,
```

Comment:
This should be an `InetSocketAddress`.  Also we will need to preserve the old constructor to not break things, which means you should add a new one (package-private)

Reply:
One question, Why `InetSocketAddress` is better than `String`, It's the reason the `InetSocketAddress` are more reliable?

Reply:
And the InetSocketAddress contains the port config?

Reply:
@Shoothzj Yes, `InetSocketAddress` contains IP Address and Port.